src/type_.rs

```diff
@@ -132,8 +137,13 @@ impl<'gcc, 'tcx> BaseTypeMethods<'tcx> for CodegenCx<'gcc, 'tcx> {
         self.double_type
```

Comment:
Depending on what you decide to do above, you could return `Float32`/`Float64` in those functions if you decide to add the support for checking if those types are supported.

Of course, we would keep returing `Float` and `Double` on platforms where those types are not supported.